backend/controllers/addGroup.js

```diff
@@ -0,0 +1,56 @@
+const { serviceUnavailableMessage } = require('../common-messages')
+
+const selectGroupFactory = (getUserAndGroups, saveGroup, maintainUrl, maintainTitle, logError) => {
+  const stashStateAndRedirectToIndex = (req, res, errors) => {
+    req.flash('addGroupErrors', errors)
+    res.redirect(req.originalUrl)
+  }
+
+  const index = async (req, res) => {
+    const { username } = req.params
+    const staffUrl = `${maintainUrl}/${username}`
+
+    try {
+      const [assignableGroups, user] = await getUserAndGroups(res.locals, username)
+      const groupDropdownValues = assignableGroups.map((g) => ({
+        text: g.groupName,
+        value: g.groupCode,
+      }))
+
+      res.render('addGroup.njk', {
+        maintainTitle,
+        maintainUrl,
+        staff: { username: user.username, name: `${user.firstName} ${user.lastName}` },
+        staffUrl,
+        groupDropdownValues,
+        errors: req.flash('addGroupErrors'),
+      })
+    } catch (error) {
+      logError(req.originalUrl, error, serviceUnavailableMessage)
+      res.render('error.njk', { url: staffUrl })
+    }
+  }
+
+  const post = async (req, res) => {
+    const { username } = req.params
+    const { group } = req.body
+    const staffUrl = `${maintainUrl}/${username}`
+
+    try {
+      if (!group) {
+        const errors = [{ href: '#group', text: 'Select a group' }]
```

Comment:
Meh, prob doesn't need to be protected by a try catch. Shame you couldn't just do the following before the try catch:

` if(!group) return stashStateAndRedirectToIndex(req, res, errors)`